test/compiler/EscapeAnalysis/EAUtils.jl

```diff
@@ -192,51 +188,43 @@ function run_passes_ipo_safe_with_ea(interp::EscapeAnalyzer,
     @timeit "slot2reg"  ir = slot2reg(ir, ci, sv)
     # TODO: Domsorting can produce an updated domtree - no need to recompute here
     @timeit "compact 1" ir = compact!(ir)
-    nargs = let def = sv.linfo.def; isa(def, Method) ? Int(def.nargs) : 0; end
-    local state
-    if is_ipo_profitable(ir, nargs) || caller.linfo.specTypes === interp.entry_tt
-        try
-            @timeit "[IPO EA]" begin
-                state = analyze_escapes(ir, nargs, false, get_escape_cache(interp))
-                cache_escapes!(interp, caller, state, cccopy(ir))
-            end
-        catch err
-            @error "error happened within [IPO EA], inspect `Main.ir` and `Main.nargs`"
-            @eval Main (ir = $ir; nargs = $nargs)
-            rethrow(err)
-        end
-    end
-    if caller.linfo.specTypes === interp.entry_tt && !interp.optimize
-        # return back the result
-        interp.ir = cccopy(ir)
-        interp.state = state
-        interp.mi = sv.linfo
-    end
     @timeit "Inlining"  ir = ssa_inlining_pass!(ir, sv.inlining, ci.propagate_inbounds)
     # @timeit "verify 2" verify_ir(ir)
     @timeit "compact 2" ir = compact!(ir)
-    if caller.linfo.specTypes === interp.entry_tt && interp.optimize
-        try
-            @timeit "[Local EA]" state = analyze_escapes(ir, nargs, true, get_escape_cache(interp))
-        catch err
-            @error "error happened within [Local EA], inspect `Main.ir` and `Main.nargs`"
-            @eval Main (ir = $ir; nargs = $nargs)
-            rethrow(err)
-        end
-        # return back the result
-        interp.ir = cccopy(ir)
-        interp.state = state
-        interp.mi = sv.linfo
-    end
     @timeit "SROA"      ir = sroa_pass!(ir, sv.inlining)
     @timeit "ADCE"      ir = adce_pass!(ir, sv.inlining)
     @timeit "compact 3" ir = compact!(ir, true)
     if JLOptions().debug_level == 2
         @timeit "verify 3" (verify_ir(ir); verify_linetable(ir.linetable))
     end
+    nargs = let def = sv.linfo.def; isa(def, Method) ? Int(def.nargs) : 0; end
+    get_escape_cache = GetEscapeCache(interp)
+    local estate::EscapeState
+    try
+        @timeit "EA" estate = analyze_escapes(ir, nargs, get_escape_cache)
+    catch err
+        @error "error happened within EA, inspect `Main.ir` and `Main.nargs`"
+        @eval Main (ir = $ir; nargs = $nargs)
```

Comment:
Maybe while we keep this debugging logic around, we should give these more explanatory names?

```suggestion
        @error "error happened within EA, inspect `Main.failed_escapeanalysis_ir` and `Main.failed_escapeanalysis_nargs`"
        @eval Main (failed_escapeanalysis_ir = $ir; failed_escapeanalysis_nargs = $nargs)
```